_posts/2015-07-15-04-14-capabilities-vocabulary-support.md

```diff
@@ -0,0 +1,168 @@
+
+	[NotFilterable]
+	[NotSortable]
+	public Guid Token { get; set; }
+
+	[NotNavigable]
+	public string Email { get; set; }
+
+	[NotCountable]
+	public IList<Address> Addresses { get; set; }
+
+	[NotCountable]
+	public IList<Color> FavoriateColors { get; set; }
+
+	[NotExpandable]
+	public IEnumerable<Order> Orders { get; set; }
+}
+{% endhighlight %}
+
+Where, `Address` is a normal complex type, `Color` is an enum type and `Order` is a normal entity type. You can find their definitions in the sample codes.
+
+Based on the above CLR classes, we can build the Edm model as:
+{% highlight csharp %}
+private static IEdmModel GetEdmModel()
+{
+	var builder = new ODataConventionModelBuilder();
+	builder.EntitySet<Customer>("Customers");
+	builder.EntitySet<Order>("Orders");
+	return builder.GetEdmModel();
+}
+{% endhighlight %}
+
+### Expose annotations
+
+Now, you can query the metadata document for capabilites vocabulary annotation as:
+
+{% highlight xml %}
+<?xml version="1.0" encoding="utf-8"?>
+<edmx:Edmx Version="4.0" xmlns:edmx="http://docs.oasis-open.org/odata/ns/edmx">
+  <edmx:DataServices>
+    <Schema Namespace="CapabilitiesVocabulary" xmlns="http://docs.oasis-open.org/odata/ns/edm">
+      <EntityType Name="Customer">
+        <Key>
+          <PropertyRef Name="CustomerId" />
+        </Key>
+        <Property Name="CustomerId" Type="Edm.Int32" Nullable="false" />
+        <Property Name="Name" Type="Edm.String" />
+        <Property Name="Token" Type="Edm.Guid" Nullable="false" />
+        <Property Name="Email" Type="Edm.String" />
+        <Property Name="Addresses" Type="Collection(CapabilitiesVocabulary.Address)" />
+        <Property Name="FavoriateColors" Type="Collection(CapabilitiesVocabulary.Color)" Nullable="false" />
+        <NavigationProperty Name="Orders" Type="Collection(CapabilitiesVocabulary.Order)" />
+      </EntityType>
+      <EntityType Name="Order">
+        <Key>
+          <PropertyRef Name="OrderId" />
+        </Key>
+        <Property Name="OrderId" Type="Edm.Int32" Nullable="false" />
+        <Property Name="Price" Type="Edm.Double" Nullable="false" />
+      </EntityType>
+      <ComplexType Name="Address">
+        <Property Name="City" Type="Edm.String" />
+        <Property Name="Street" Type="Edm.String" />
+      </ComplexType>
+      <EnumType Name="Color">
+        <Member Name="Red" Value="0" />
+        <Member Name="Green" Value="1" />
+        <Member Name="Blue" Value="2" />
+        <Member Name="Yellow" Value="3" />
+        <Member Name="Pink" Value="4" />
+        <Member Name="Purple" Value="5" />
+      </EnumType>
+    </Schema>
+    <Schema Namespace="Default" xmlns="http://docs.oasis-open.org/odata/ns/edm">
+      <EntityContainer Name="Container">
+        <EntitySet Name="Customers" EntityType="CapabilitiesVocabulary.Customer">
+          <NavigationPropertyBinding Path="Orders" Target="Orders" />
+          <Annotation Term="Org.OData.Capabilities.V1.CountRestrictions">
```

Comment:
Highlight the capability annotation added to the $metadata.  Add a link to: http://docs.oasis-open.org/odata/odata/v4.0/errata02/os/complete/vocabularies/Org.OData.Capabilities.V1.xml